### PR TITLE
[C#] allow customization of generated enum suffixes

### PR DESCRIPTION
**Pull request by scottdallamura**
allows customization of enum suffixes for C#
addresses #4275 

```
{
    "apiPackage": "RestApi.Controllers",
    "modelPackage": "RestApi.Models",
    "packageName": "RestApi",
    "enumNameSuffix": "aaa",
    "enumValueNameSuffix": "bbb"
}
```
generates stuff like (not that i would ever want 'aaa' or 'bbb' suffixes, but it's a fine example...)
```
        /// <summary>
        /// Gets or Sets Status
        /// </summary>
        [TypeConverter(typeof(CustomEnumConverter<Statusaaa>))]
        [JsonConverter(typeof(Newtonsoft.Json.Converters.StringEnumConverter))]
        public enum Statusaaa
        {
            
            /// <summary>
            /// Enum Activebbb for Active
            /// </summary>
            [EnumMember(Value = "Active")]
            Activebbb = 1,
            
            /// <summary>
            /// Enum Disabledbbb for Disabled
            /// </summary>
            [EnumMember(Value = "Disabled")]
            Disabledbbb = 2
        }
```

### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) before.
- [x] Run the shell script(s) under `./bin/` (or Windows batch scripts under`.\bin\windows`) to update Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit, and these must match the expectations made by your contribution. You only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the code or mustache templates for a language (`{LANG}`) (e.g. php, ruby, python, etc).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.1.x`, `5.0.x`. Default: `master`.
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### CODE REVIEW

**Comment by scottdallamura** [comment]
copying the technical committee ...
@mandrean @jimschubert @frankyjuang

**Comment by scottdallamura** [comment]
so i think i figured out how to run the scripts, and indeed a bunch of files were generated.  i expected the aspnetcore one (and i've committed the updated version) but it's unclear to me why so many samples for other languages were generated. i'm trying it without my changes now.

**Comment by scottdallamura** [comment]
looks ready to go? let's do it

**Comment by scottdallamura** [comment]
oops. fixed it!

File: docs/generators/aspnetcore.md [hunk 1]
```diff
@@ -27,6 +27,8 @@ sidebar_label: aspnetcore
 |useNewtonsoft|Uses the Newtonsoft JSON library.| |true|
 |newtonsoftVersion|Version for Microsoft.AspNetCore.Mvc.NewtonsoftJson for ASP.NET Core 3.0+| |3.0.0-preview5-19227-01|
 |useDefaultRouting|Use default routing for the  ASP.NET Core version. For 3.0 turn off default because it is not yet supported.| |true|
+|enumNameSuffix|Suffix that will be appended to all enum names.| |Enum|
+|enumValueNameSuffix|Suffix that will be appended to all enum value names.| |Enum|
```
Comment (jimschubert) [inline docs/generators/aspnetcore.md]: What is the use case for this? Why wouldn't someone just put the expected value in the enum text field in the specification document?

Comment (scottdallamura) [reply docs/generators/aspnetcore.md]: The behavior before my change is to append `Enum` to both the name of the enum type and each value.  So you end up with
```
enum MyEnum {
  Value1Enum,
  Value2Enum
}
```
I didn't want to break the existing default behavior, so I made that suffix customizable too.

Comment (jimschubert) [reply docs/generators/aspnetcore.md]: Sorry for the delay in response. The holidays were pretty hectic for me.

I think we'll want to call the value one `enumValueSuffix`. I'll do that in a separate commit.

**Pull request merged**